### PR TITLE
Set mandatory support phone

### LINKED ISSUE #9556: Set support phone number's initial value from onboarding phone number
### Description

The support phone number field on the Payment settings page defaults to empty for newly onboarded accounts. When the merchant does not provide its value, support requests are routed to the platform's support phone number which leads to issues described at pbUcTB-5ch-p2

Image

### Acceptance criteria

Set this field using the phone number entered on the onboarding form.

Image

### Dev notes

See my comment at pbUcTB-5ch-p2#comment-11620.

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #9556 

#### Changes proposed in this Pull Request

The support phone number field on the Payment settings page defaults to empty for newly onboarded accounts. In this PR the support phone is set to mandatory, so the user is forced to set the phone number.

#### Testing instructions

* onboard new account
* go to `WooCommerce -> Settings` page
* an error should be displayed if the phone field is empty

Screenshot 2024-11-20 at 13 29 23

* the "Save changes" button should NOT be active until we have the support phone number is set

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9766` or branch name `add/9556-set-support-phone-mandatory` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/9556-set-support-phone-mandatory) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 09ea2c75c80ba9b5cf6e9953a987df99ceda7166
- Build time: 2024-11-25 08:09:59 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/support-phone-input/index.js [hunk 1]
```diff
@@ -28,6 +28,12 @@ const SupportPhoneInput = ( { setInputVallid } ) => {
 		isTestModeOnboarding && supportPhone === '+10000000000';
 
 	const [ isPhoneValid, setPhoneValidity ] = useState( true );
+	if ( supportPhone === '' ) {
+		supportPhoneError = __(
+			'Support phone cannot be empty.',
```
Comment (dmallory42) [inline client/settings/support-phone-input/index.js]: Nitpick - maybe use `Support phone number cannot be empty.` as the message here?

**Review by oaratovskyi — APPROVED** [review]
LGTM and works well! :shipit:

**Review by mordeth — APPROVED** [review]
✅ Code changes are looking good!
✅ Manually tested and everything works as expected.

**Pull request merged**